components/collector/src/source_collectors/gitlab/pipeline_duration.py

```diff
@@ -12,10 +12,21 @@ class GitLabPipelineDuration(GitLabPipelineBase):
 
     async def _parse_value(self, responses: SourceResponses, included_entities: Entities) -> Value:
         """Parse the value from the responses."""
-        if durations := [int(entity["duration"]) for entity in included_entities]:
-            if self._parameter("pipeline_selection") == "slowest":
-                return str(max(durations))
-            included_entities.sort(key=lambda entity: entity["updated"] or entity["created"])
-            return str(included_entities[-1]["duration"])
-        error_message = "No pipelines found within the lookback period"
-        raise CollectorError(error_message)
+        if len(included_entities) == 0:
+            error_message = "No pipelines found within the lookback period"
+            raise CollectorError(error_message)
+        match self._parameter("pipeline_selection"):
+            case "slowest":
+                return str(max(self._durations(included_entities)))
+            case "latest":
+                included_entities.sort(key=lambda entity: entity["updated"] or entity["created"])
```

Comment:
I see that this was the old logic, but can an entity which has `created` but not `updated` actually have a duration?